MzLibUtil/Datum.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Text;
+
+namespace MzLibUtil
+{
+    public class Datum
+    {
+        public readonly double X;
+        public readonly double? Y;
+        public readonly double? Z;
+        public readonly double Weight;
+        public readonly string Label;
+
+        public Datum(double x, double? y = null, double? z = null, string label = null, double weight = 1)
+        {
+            this.X = x;
+            this.Y = y;
+            this.Z = z;
+
+            this.Label = label;
+            this.Weight = weight;
+        }
+
+        public override string ToString()
```

Comment:
Should this class have an "Equals" methods?

Reply:
this one i'm not sure about... part of me says they're distinct data points even if they happen to have equal data values, another part says that if they have the same data they're equal to each other..

Reply:
having an equals doens't require the user to do anything in response. its simply a comparer.

Reply:
it could affect downstream usage somewhat weirdly though. like if you do:

```cs
List<Datum> myListOfDataPoints = new List<Datum> { new Datum(1, 0), new Datum(1, 0) };
var distinctDataPoints = myListOfData.Distinct();

bool test = distinctDataPoints.Contains(new Datum(1, 0));
```

`distinctDataPoints` would only have 1 item in it, since the Datum objects in myListOfDataPoints are considered duplicates. `test` would also return `true`. Which I'm not sure is "correct" behavior. I'm not really sure of a situation where you'd ever want to get distinct data points though, so I'm not sure what the "correct" behavior should be